File: ruby_solutions/problem_4_largest_palindrome_product.rb
#A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

#Find the largest palindrome made from the product of two 3-digit numbers.

def palindrome?(integer)
  integer.to_s.reverse.to_i == integer ? true : false
end

def three_digit_product
  palindrome = []
  range = *(100..999)
  range.each do |num|
    range.each do |int|
      if palindrome?(num*int)
        palindrome << (num*int)
      end
    end
  end
  return palindrome.sort.last
end

p three_digit_product

File: javascript_solutions/problem_4_largest_palindrome_product.js
// A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

// Find the largest palindrome made from the product of two 3-digit numbers.

var products = []

function run() {
  for (var i = 100; i < 1000; i++) {
    for (var j = 100; j < 1000; j++) {
      var product = i*j
      var original = product.toString();
      var reversed = product.toString().split("").reverse().join("");
      if (original == reversed) {
        products.push(i*j)
      }
    }
  }
}

run();

console.log(products.sort(function(a,b) { return a - b}))